### PR TITLE
Add evaluate to user endpoints

### PR DESCRIPTION
**Pull request by jharini**

### CODE REVIEW

File: fusillade/api/users/__init__.py [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
 from fusillade import User, directory
+from fusillade.utils.authorize import assert_authorized
 
 
-def put_new_user():
+def put_new_user(token_info:dict):
     json_body = request.json
+    assert_authorized(token_info['https://auth.data.humancellatlas.org/email'],
+                      ['fus:PutUser'],
+                      [f'arn:hca:fus:*:*:user/{json_body["user_id"]}/'])
```
Comment (Bento007) [inline fusillade/api/users/__init__.py]: `arn:hca:fus:*:*:user/` seems more appropriate, since the user/{user_id} resource doesn't exist yet.

Comment (Bento007) [reply fusillade/api/users/__init__.py]: also is the trailing `/` needed?

Comment (jharini) [reply fusillade/api/users/__init__.py]: updated

File: fusillade/api/users/__init__.py [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
     user.add_roles(json_body.get('roles', []))
     user.add_groups(json_body.get('groups', []))
     return make_response('', 201)
 
-
-def get_user(user_id):
+def get_user(token_info: dict, user_id: str):
+    assert_authorized(token_info['https://auth.data.humancellatlas.org/email'],
+                      ['fus:GetUser'],
+                      [f'arn:hca:fus:*:*:user/{user_id}/'])
```
Comment (Bento007) [inline fusillade/api/users/__init__.py]: remove trailing `/`

**Pull request merged**